lib/train/transports/ssh_connection.rb

```diff
@@ -285,7 +285,12 @@ def execute_on_channel(cmd, &data_handler)
       exit_status = nil
       session.open_channel do |channel|
         # wrap commands if that is configured
-        cmd = @cmd_wrapper.run(cmd) unless @cmd_wrapper.nil?
+        if !@cmd_wrapper.nil?
+          cmd = @cmd_wrapper.run(cmd)
+        elsif @transport_options[:sudo] || @transport_options[:shell]
+          # if sudo or shell is in the options we can assume Linux/Unix
+          cmd = LinuxCommand.new(self, @transport_options).run(cmd)
```

Comment:
For other reviewers, your choices are LinuxCommand (which implements sudo facilities), WindowsCommand, or a shared superclass. LinuxCommand is too tightly named, but this PR appears to be using it correctly.

Reply:
Effectively this is what other code parts supply, don't want to change lots of other code but yeah, "LinuxCommand" covers all kind of Linux/Unix OS so it should be probably named "UnixCommand".